src/selfie_intersection/src/intersection_server.cpp

```diff
@@ -99,16 +106,23 @@ void IntersectionServer::intersection_callback(const std_msgs::Float32 &msg)
 {
   point_min_x_ = msg.data;
   point_max_x_ = point_min_x_ + road_width_;
-  ROS_INFO_THROTTLE(1, "Distance to intersection: %lf", point_min_x_);
+  if (intersectionServer_.isActive())
+    ROS_INFO_THROTTLE(1, "Distance to intersection: %lf", point_min_x_);
+  if (visualization_ && intersectionServer_.isActive())
+  {
+    Box(point_min_x_, point_max_x_, point_min_y_, point_max_y_)
+        .visualize(visualize_intersection_, "area_of_interest", 0.9, 0.9, 0.1, 0);
+  }
```

Comment:
Jeśli nie wyhamujemy i zdarzak zasłoni kamerze linię na skrzyżowaniu nie będzie już tutaj wiadomości, czyli nie będzie wizualizacji AOI. Przeniósłbym do callbacka od lidaru i dał lifetime na jakieś 0.2 (lidar publikuje co 0,1s + 0,1 dla pewności)

Reply:
Przeniosę, ale wolę zostawić dłuższy lifetime (np 3 s) w razie gdyby coś się posypało, albo nagle akcja się zakończyła to wtedy znaczniki zostaną na tyle długo abyśmy zdążyli zobaczyć co poszło źle.